src/View/Helper/DocBlockHelper.php

```diff
@@ -1,14 +1,52 @@
 <?php
 namespace Bake\View\Helper;
 
+use Cake\Collection\Collection;
 use Cake\ORM\Association;
+use Cake\Utility\Hash;
 use Cake\View\Helper;
 
 /**
  * DocBlock helper
  */
 class DocBlockHelper extends Helper
 {
+    protected $_annotationSpacing = true;
+
+    /**
+     * Writes the DocBlock header for a class which includes the property and method declarations. Annotations are
+     * sorted and grouped by type and value. Groups of annotations are separated by blank lines.
+     *
+     * @param string $className The class this comment block is for.
+     * @param string $classType The type of class (example, Entity)
+     * @param array $annotations An array of PHP comment block annotations.
+     * @return string The DocBlock for a class header.
+     */
+    public function classDescription($className, $classType, array $annotations)
+    {
+        $lines = ["{$className} {$classType}", ""];
+
+        $previous = false;
+        foreach ($annotations as $ann) {
+            if (strlen($ann) > 1 && $ann[0] == '@' && strpos($ann, ' ') > 0) {
+                $type = substr($ann, 0, strpos($ann, ' '));
+                if ($this->_annotationSpacing &&
+                    $previous !== false &&
+                    $previous != $type
+                ) {
+                    $lines[] = '';
+                }
+                $previous = $type;
+            }
+            $lines[] = $ann;
+        }
+
+        $lines = array_merge(["/**"], (new Collection($lines))->map(function ($line) {
+            return rtrim(" * {$line}");
+        })->toArray(), [" */"]);
+        return implode(PHP_EOL, $lines);
```

Comment:
@markstory I thought using `PHP_EOL` was standard practice? I can change it to `"\n"` if you want.